analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DisposeFromDispose.CSharp7_2.cs

```diff
@@ -52,4 +53,99 @@ public void Dispose()
         }
     }
 
+    public class OutterClass : IDisposable
+    {
+        private Stream stream;
+
+        class InnerClass : IDisposable
+        {
+            public void M(OutterClass outter)
+            {
+                outter.stream.Dispose(); // Compliant
+            }
+
+            public void Dispose() { }
+        }
+
+        public void Dispose() { }
+    }
+
+    public class BaseClass : IDisposable
+    {
+        protected Stream stream;
+
+        public void Dispose() { }
+
+    }
+
+    public class Derived : BaseClass, IDisposable
+    {
+        public void Cleanup() => stream.Dispose(); // Compliant
+    }
+
+    public class Conditional: IDisposable
+    {
+        private Stream fs;
+
+        private void MemberBinding()
+        {
+            fs?.Dispose();                             // Noncompliant {{Move this 'Dispose' call into this class' own 'Dispose' method.}}
+        }
+
+        private void ThisMemberBinding()
+        {
+            this.fs?.Dispose();                        // Noncompliant
+        }
+
+        private void ThisAndMemberBinding()
+        {
+            this?.fs?.Dispose();                       // Noncompliant
+        }
+
+        private void Ternary()
+        {
+            (fs == null ? null : fs)?.Dispose();       // Compliant
+        }
```

Comment:
I assume there is a good reason, but can you explain to me, why this is compliant?

Reply:
This ternary is a placeholder for any kind of more complex expression. You often can not find the referenced field fs in any expression that is more complex than a simple name.
In other rules, we are more aggressive and descend into the expression and look for "simple name fs" in any of the children but this comes with its own set of problems: e.g. `(fs == null ? someStream : someOtherStream).Dispose()`.